### PR TITLE
Change creation of repository and persister

### PR DESCRIPTION
**Pull request by ValeriiSudakov**
- Changed std::function TaskRepository Factory-method to Factory class. Changed implementation in the system.
- Renamed Persister Factory method.

### CODE REVIEW

**Review by IlliaDmitriiev — APPROVED** [review]

File: core/Memory_Model/Storage/TaskRepositoryController.h [hunk 1]
```diff
@@ -5,12 +5,13 @@
 #ifndef TASKMANAGER_CORE_MEMORY_MODEL_STORAGE_TASKREPOSITORYCONTOLLER_H_
 #define TASKMANAGER_CORE_MEMORY_MODEL_STORAGE_TASKREPOSITORYCONTOLLER_H_
 #include "TaskRepository.h"
+#include "Memory_Model/RepositoriesFactory/RepositoriesFactory.h"
 
 class TaskRepositoryController {
```
Comment (denys-hrulov) [inline core/Memory_Model/Storage/TaskRepositoryController.h]: You need virtual functions and destructor to provide any kind of testability and Dependency Injection.

**Review by denys-hrulov — CHANGES_REQUESTED** [review]

File: main.cpp [hunk 2]
```diff
@@ -3,15 +3,19 @@
...
 #include "States/StatesID.h"
 #include "Memory_Model/RepositoriesFactory/TaskRepositoryFactory.h"
-
+#include "Memory_Model/Storage/TaskRepositoryController.h"
 #include "Factory/Factory.h"
 
 int main(){
   std::unique_ptr<RepositoriesFactory> factory = std::make_unique<TaskRepositoryFactory>();
-  std::unique_ptr<TaskService> service = std::make_unique<TaskServiceClass>(std::move(factory));
+  std::unique_ptr<RepositoryController> controller = std::make_unique<TaskRepositoryController>(std::move(std::move(factory)));
```
Comment (denys-hrulov) [inline main.cpp]: Cannot understand why double move std::move(std::move(...)).

File: tests/core/TestTaskService.cpp [hunk 3]
```diff
@@ -8,13 +8,14 @@
...
 #include "Memory_Model/RepositoriesFactory/TaskRepositoryFactory.h"
-
+#include "Memory_Model/Storage/TaskRepositoryController.h"
 class TestTaskService : public ::testing::Test {
 
  protected:
   virtual void SetUp() {
     std::unique_ptr<RepositoriesFactory> factory = std::make_unique<TaskRepositoryFactory>();
-    ts = std::make_unique<TaskServiceClass>(std::move(factory));
+    std::unique_ptr<RepositoryController> controller = std::make_unique<TaskRepositoryController>(std::move(std::move(factory)));
```
Comment (denys-hrulov) [inline tests/core/TestTaskService.cpp]: Same thing

File: tests/core/TestTaskService.cpp [hunk 4]
```diff
@@ -7,12 +7,15 @@
...
-
+#include "Memory_Model/RepositoriesFactory/TaskRepositoryFactory.h"
+#include "Memory_Model/Storage/TaskRepositoryController.h"
 class TestTaskService : public ::testing::Test {
 
  protected:
   virtual void SetUp() {
-    ts = std::make_unique<TaskServiceClass>(TaskServiceUtils::GetRepositoryFactory());
+    auto factory = std::make_unique<TaskRepositoryFactory>();
+    auto controller = std::make_unique<TaskRepositoryController>(std::move(std::move(factory)));
```
Comment (denys-hrulov) [inline tests/core/TestTaskService.cpp]: What the f* ?

**Review by denys-hrulov — APPROVED** [review]

**Pull request merged**